src/H5Rint.c

```diff
@@ -746,8 +746,10 @@ H5R__copy(const H5R_ref_priv_t *src_ref, H5R_ref_priv_t *dst_ref)
         dst_ref->info.obj.filename = NULL;
 
         /* Set location ID and hold reference to it */
-        if (H5R__set_loc_id(dst_ref, src_ref->loc_id, TRUE, TRUE) < 0)
```

Comment:
H5R__set_loc_id checks the destination reference's loc_id field and decrements it if it is set to a valid ID before it assigns a new loc_id to the destination reference. After discussing this with @soumagne, it seems we should not be checking anything about the destination reference since its storage is probably uninitialized.